### PR TITLE
Release 1.4.0.rc1

### PR DESCRIPTION
**Pull request by SpComb**
This PR only includes the `VERSION` bump + Changelog for the v1.4 branch. It is intended to be squashed into the v1.4 branch once the pre release is RC-complete.

* [Changelog](https://github.com/kontena/kontena/blob/release/v1.4.0.rc1/CHANGELOG.md#140rc1-2017-09-14)

## TODO
- [x] Write up changelog release highlights
- [x] #2802 Breaking changes: no more `to: env` by default, use `from: variable`
- [x] #2701 Breaking changes: `post_start` hooks are executed on each container start, not just on create

    This includes  `kontena service start` and process exit/crash -> container (re)start, but does not include `kontena service restart` or healthcheck restarts.

- [x] #2701 changelogs (stack hooks)

- [ ] Optional agent `wss://...` SSL certificate validation?

- [ ] CLI omnibus deb packages? CLI Windows `gem install` with the `websocket-driver` native extensions?

## Waiting for fixes to open blocker issues
- [x] #2815 Stack reader does not allow interpolating `to: env: FOO` using `${FOO}`
- [x] #2851 DB migration failure on HostNode indexes for 1.3.4 -> 1.4.0.pre
- [x] #2855 server checks agent websocket Kontena-Node-Name before version check

## Waiting for remaining open feature/bugfix PRs to be merged
- [x] #2511 Enhanced HostNode status, health
- [x] #2802 Do not send variables to env by default
- [x] #2816 Certificate auto renewal
- [x] #2708 Fix exec console size
- [x] #2767 Support shm_size
- [x] #2791 Migrate restart-policy containers
- [x] #2701 Stack: pre_start/stop hooks

### CODE REVIEW

**Comment by SpComb** [comment]
Updated for 1.4.0.pre9

**Comment by SpComb** [comment]
Four types of e2e spec failures:

* e2e stack ugprade specs fail on the new confirm prompts #2879
* kontena stack validate --online fails on GRID/STACK/PLATFORM variables #2880
* kontena plugin uninstall does not prompt or accept --force #2810
* e2e plugin support specs: Could not find gem 'dotenv' in any of the gem sources listed in your Gemfile. #2811

None of these look like RC blockers, although #2880 looks like a legit bug. Not sure about #2810 yet, it's documented in the changelogs as-is.

```
Failed examples:

rspec ./spec/features/plugin/run_spec.rb:5 # plugin support spinner is available by including Kontena::Cli::ShellSpinner
rspec ./spec/features/plugin/run_spec.rb:11 # plugin support spinner is available by including Kontena::Cli::Common
rspec ./spec/features/plugin/run_spec.rb:17 # plugin support spinner is available without including anything
rspec ./spec/features/plugin/run_spec.rb:25 # plugin support client does not need be required
rspec ./spec/features/plugin/run_spec.rb:33 # plugin support config does not need to be required
rspec ./spec/features/plugin/run_spec.rb:41 # plugin support common has access to the essentials
rspec ./spec/features/plugin/uninstall_spec.rb:8 # plugin uninstall removes installed plugin
rspec ./spec/features/stack/upgrade_spec.rb:31 # stack upgrade from file prompts if the stack is different
rspec ./spec/features/stack/upgrade_spec.rb:57 # stack upgrade for a stack that is linked to externally fails to upgrade if linked
rspec ./spec/features/stack/upgrade_spec.rb:68 # stack upgrade for a stack that is linked to externally fails to deploy if linked
rspec ./spec/features/stack/validate_spec.rb:7 # stack validate service_link allows to select matching service
```

**Comment by SpComb** [comment]
The #2879 spec failures covered up #2883, which is a legit bug and probably RC-critical (`kontena stack install` fails with stack dependencies).

**Comment by SpComb** [comment]
Updated for v1.4.0.pre11

@jakolehm PTAL at the changelog?

**Comment by jakolehm** [comment]
Changelog LGTM.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,508 @@
...
+
+    The `kontena * logs` and `kontena * events` commands have been changed to to use `--follow` `--tail=N` options.
+
+* `kontena stack install -v foo=bar`
+
+    Provide values for stack variables when installing, skipping any prompts. Similar to `--values-from`.
+
+    Use `-v foo.var=bar` to provide values for the `depends: foo` child stack `variables`.
+
+* `kontena stack install/upgrade --values-from-stack`
```
Comment (kke) [inline CHANGELOG.md]: `kontena stack [install|upgrade|build|validate] --values-from-stack <stack_name>`

**Comment by SpComb** [comment]
Updated for pre12

**Comment by SpComb** [comment]
Updated for pre13; waiting for the CLI omnibus deploy to succeed.

**Comment by SpComb** [comment]
Updated for pre14

**Comment by SpComb** [comment]
1.4.0.pre14 checks out with a quick pre11 upgrade test, so this should be ready to go now.

**Comment by SpComb** [comment]
Updated the changelog... not waiting for re-review, we will still edit it for the final release.

**Pull request merged**